src/platform_impl/macos/window.rs

```diff
@@ -122,10 +122,17 @@ fn create_window(
         let frame = match screen {
             Some(screen) => appkit::NSScreen::frame(screen),
             None => {
-                let (width, height) = attrs
-                    .inner_size
-                    .map(|logical| (logical.width, logical.height))
-                    .unwrap_or_else(|| (800.0, 600.0));
+                // Not sure what dpi_factor we need here.
+                // If i understand correctly, here we fail to create NSScreen frame, so we've got
+                // no dpi_factor
```

Comment:
This code path determines the initial dimensions of the window when it's not in fullscreen mode, from the size provided by the user, or if not provided, a default value of 800 by 600 logical pixels is used. So this would incorrectly use a DPI factor of 1.0 for the dimensions provided by the user. Perhaps we can defer setting the window frame until the `NSWindow` is initialized and the appropriate `backingScaleFactor` can be used?

Reply:
The trouble is we need a `frame` to initialise an `NSWindow`. Also in this branch of `match` `screen` is `None`, so we cannot query it for `backingScaleFactor`.
The `screen` being `None` is kind of odd, by the way.

Reply:
I think we can pick the screen with [`init:contentRect:styleMask:backing:defer:screen:`](https://developer.apple.com/documentation/appkit/nswindow/1419755-init) for the window. We could set this to the main screen (i.e. the screen that has the current keyboard focus, `NSScreen:main`). We would then be able to use the `backingScaleFactor` of the main screen here.